File: jdi-uitest-unittests/src/test/java/com/epam/jdi/uitests/testing/unittests/InitTests.java
package com.epam.jdi.uitests.testing.unittests;

import com.epam.jdi.uitests.testing.unittests.entities.User;
import com.epam.jdi.uitests.testing.unittests.pageobjects.EpamJDISite;
import com.epam.jdi.uitests.web.testng.testRunner.TestNGBase;
import com.epam.web.matcher.verify.Verify;
import org.testng.annotations.AfterMethod;
import org.testng.annotations.BeforeSuite;

import static com.epam.jdi.uitests.testing.unittests.pageobjects.EpamJDISite.homePage;
import static com.epam.jdi.uitests.testing.unittests.pageobjects.EpamJDISite.login;
import static com.epam.jdi.uitests.web.WebSettings.*;
import static com.epam.jdi.uitests.web.selenium.elements.composite.Site.Init;


/**
 * Created by Roman_Iovlev on 7/13/2015.
 */
public class InitTests extends TestNGBase {

    @BeforeSuite(alwaysRun = true)
    public static void setUp() throws Exception {
        getDriverFactory().setDriverPath("C:\\Selenium");
        initJDIFromProperties();
        //Assert.noScreenOnFail();
        Init(EpamJDISite.class);
        homePage.open();
        login.submit(User.DEFAULT);
        logger.info("Run Tests");
        Verify.getFails();
    }

    @AfterMethod
    public void tearDown() {
        Verify.getFails();
    }
}

File: Python/Test/jdi_uitests_webtests/test/init_tests.py
import unittest

from JDI.web.selenium.settings.web_settings import WebSettings
from JDI.web.selenium.elements.composite.web_site import WebSite
from Test.jdi_uitests_webtests.main.entities.user import User
from Test.jdi_uitests_webtests.main.page_objects.epam_jdi_site import EpamJDISite
from Test.jdi_uitests_webtests.main.page_objects.w3c_site.w3c_site import W3cSite


class InitTests(unittest.TestCase):

    @classmethod
    def setUpClass(cls):
        WebSite.init(EpamJDISite, "chrome")
        WebSettings.logger.info("\nRun Tests from '%s' file" % cls.__name__)
        EpamJDISite.home_page.open()
        EpamJDISite.login_page.submit(User.default())

    @classmethod
    def setUp(self, name=""):
        WebSettings.logger.info("\nRun Test '%s'" % name)

    @classmethod
    def tearDownClass(cls):
        try:
            WebSettings.quit_browser()
        except: pass


class W3CInit(unittest.TestCase):

    @classmethod
    def setUpClass(cls):
        WebSite.init(W3cSite, "chrome")
        WebSettings.logger.info("Run Tests")
        W3cSite.frame_page.open()


    @classmethod
    def tearDownClass(cls):
        WebSettings.quit_browser()